llmcord/llm_tools.py

```
@@ -0,0 +1,185 @@
+        return "Error: No query provided."
+
+    logger.info("Performing web search for: {}", query)
+    results: list[str] = []
+    try:
+        # The latest DDGS version works best as a context manager
+        with DDGS() as ddgs:
+            ddgs_gen = ddgs.text(query, max_results=5)
+            for r in ddgs_gen:
+                r_dict = cast("dict[str, str]", r)
+                results.append(f"Title: {r_dict['title']}\nSnippet: {r_dict['body']}\nURL: {r_dict['href']}\n")
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)